src/history/hs.c

```diff
@@ -767,13 +788,23 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
          * tombstone.
          * 4) We have a single tombstone on the chain, it is simply ignored.
          */
+        min_insert_ts = WT_TS_MAX;
         for (non_aborted_upd = prev_upd = NULL; upd != NULL;
              prev_upd = non_aborted_upd, upd = upd->next) {
             if (upd->txnid == WT_TXN_ABORTED)
                 continue;
 
             non_aborted_upd = upd;
 
+            /*
+             * If we've seen a smaller timestamp before, use that instead.
+             *
+             * FIXME-WT-6388: Perhaps these writes need to be ordered.
+             */
+            if (min_insert_ts < upd->start_ts)
+                upd->start_ts = upd->durable_ts = min_insert_ts;
```

Comment:
Unfortunately, I think it's necessary to trample over the durable timestamp.
So this could cause some trouble if there are some resolved prepared updates thrown in the mix.

Reply:
I think this is another wrinkle to document. It would be worth creating a statistic that captures the case where `start != durable && durable < min_insert`. If a prepared transaction becomes resolved and it's durable timestamp is out of order with subsequent writes, then MongoDB has mis-used the ignore_prepare for reads and probably wants to be able to find that out because it's likely a bug.

Reply:
The stat can be added in a subsequent ticket.

Reply:
0143cd0